File: Submissions/233_Number of Digit One.cpp
class Solution {
public:
    int countDigitOne(int n) {
        int result = 0;
        int curr = n;
        unsigned int digit_number = 1;
        while (curr != 0) {
            int remainder = curr % 10;
            curr = curr / 10;
            
            if (remainder == 1) {
                result += (curr * digit_number + (n % digit_number) + 1);
            } else if (remainder > 1){
                result += (curr * digit_number + digit_number);
            } else {
                result += (curr * digit_number);
            }
            
            digit_number *=10;
        }
        return result;
    }
};

File: Kotlin/233_Number_of_Digit_One.kt
class Solution {
    fun countDigitOne(n: Int): Int {
        var result = 0
        var current = n
        var digit_number = 1
        while(current != 0) {
            val remainder = current % 10
            current /= 10
            
            result += current * digit_number
            if (remainder == 1) {
                result += (n % digit_number) + 1
            } else if (remainder > 1) {
                result += digit_number
            }
            
            digit_number *= 10
        }
        
        return result
    }
}